### PR TITLE
GRW-1981 / Feature / Report tracking events to datadog

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Tracking service reports events to Datadog as well

This is how offer_created log looks like in Datadog:
Screenshot 2022-12-29 at 11 29 29

## Justify why they are needed

More debugging info for Analytics, we may need it in the future

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: apps/store/src/services/Tracking/Tracking.ts [hunk 1]
```diff
@@ -11,22 +14,44 @@ export enum TrackingEvent {
   PageView = 'virtual_page_view',
 }
 
+datadogLogs.createLogger('tracking')
```
Comment (robinandeer) [inline apps/store/src/services/Tracking/Tracking.ts:17]: nit: use a constant to keep track the logger name

or does `createLogger` return a logger instance we can use somewhere else?

File: apps/store/src/services/Tracking/Tracking.ts [hunk 2]
```diff
@@ -11,22 +14,44 @@ export enum TrackingEvent {
...
   constructor(public context: TrackingContext = {}) {}
 
+  private logger = datadogLogs.getLogger('tracking')!
+
   public setContext = (key: string, value: unknown) => {
     if (this.context[key] !== value) {
       console.debug(`tracking context ${key}:`, value)
     }
-    this.context[key] = value
+    if (value != null) {
```
Comment (robinandeer) [inline apps/store/src/services/Tracking/Tracking.ts:29]: nit: just used to strict comparison -- shouldn't we always do that (!==)?

File: apps/store/src/services/logger/client.ts [hunk 3]
```diff
@@ -17,7 +17,11 @@ export const initDatadog = () => {
       datadogLogsConfig.beforeSend = (event) => {
         // Must exclude console origin to avoid endless loop.  Feel free to experiment with other values
         if (event.origin === 'logger') {
-          console[event.status](event.message)
+          if (event.logger?.name == 'tracking') {
```
Comment (robinandeer) [inline apps/store/src/services/logger/client.ts:20]: nit: use the constant to keep track of this string value
nit #2: use strict comparison

File: apps/store/src/services/Tracking/Tracking.ts [hunk 4]
```diff
@@ -11,22 +14,44 @@ export enum TrackingEvent {
...
   }
 
+  public setProductContext = (product: ProductData) => {
+    this.setContext('has_home', product.name.includes('_APARTMENT'))
+    this.setContext('has_house', product.name.includes('_HOUSE'))
+    this.setContext('has_card', product.name.includes('_CAR'))
+    this.setContext('has_accident', product.name.includes('_ACCIDENT'))
+    this.setContext('has_travel', false)
+    this.setContext('is_student', product.name.includes('STUDENT'))
+  }
```
Comment (robinandeer) [inline apps/store/src/services/Tracking/Tracking.ts:48]: This is starting to look very promise specific -- I'm sure we can do better ;)

But feel free to tackle it later

**Review by robinandeer — APPROVED** [review]
Great stuff -- left a bunch of comments, happy to discuss

File: apps/store/src/services/Tracking/Tracking.ts (hunk 1 above)
Comment (alebedev) [reply apps/store/src/services/Tracking/Tracking.ts:17]: The only other usage is less than 10 lines below, where we initialize `Tracking.logger`

I think adding constant does not improve readability here

File: apps/store/src/services/Tracking/Tracking.ts (hunk 2 above)
Comment (alebedev) [reply apps/store/src/services/Tracking/Tracking.ts:29]: This is intentional, called can pass null or undefined

File: apps/store/src/services/logger/client.ts (hunk 3 above)
Comment (alebedev) [reply apps/store/src/services/logger/client.ts:20]: Ok, this is another usage, I'm creating a constant

File: apps/store/src/services/Tracking/Tracking.ts (hunk 4 above)
Comment (alebedev) [reply apps/store/src/services/Tracking/Tracking.ts:48]: Sonny asked for it to be backwards compatible with current site analytics.  I don't like it, but at least we're keeping this mapping in one place

**Pull request merged**